core/orchestrator.go

```diff
@@ -2,6 +2,7 @@ package core
 
 import (
 	"context"
+	"errors"
```

Comment:
@yondonfu recently found an awesome errors package we're planning to use in our PM PRs ([link](https://github.com/pkg/errors)).

The advantage we get from using this errors package is that it captures the stack trace, which adds some very useful data to our logs. It also connects with the author's general error handling paradigm where you don't log errors as they bubble up, but instead treat them as opaque errors and only log them at the top level of handling that error (i.e. where you stop bubbling it up and actually do something with it).

Here's [one of the links](https://www.sohamkamani.com/blog/2018/05/15/golang-error-stack-trace/) Yondon shared, from the author.

@j0sh what do you think about starting to use this approach for errors?